src/shared/api.ts

```diff
@@ -0,0 +1,98 @@
+export type ApiProvider = "ollama" | "openai" | "gemini" | "anthropic";
```

Comment:
These are LLMs we are supporting and in this Ollama and Gemini are freely available. So do we need to include all other LLMs also ?

Reply:
When I try to run Ollama on my local machine its relatively slow and gemini has a request limit. So I have added two other LLM ie. OpenAI and Anthropic for us to test out the performance benifits. We can utilize Ollama or Gemini once we are done developing the extension.